controlplane/kubeadm/controllers/controller.go

```diff
@@ -580,3 +592,52 @@ func (r *KubeadmControlPlaneReconciler) adoptOwnedSecrets(ctx context.Context, k
 
 	return nil
 }
+
+func (r *KubeadmControlPlaneReconciler) remediateUnhealthy(ctx context.Context, logger logr.Logger, controlPlane *internal.ControlPlane) error {
+	if !controlPlane.RemediationAllowed() || !controlPlane.HasUnhealthyMachine() || controlPlane.Machines.Len() < int(*controlPlane.KCP.Spec.Replicas) {
```

Comment:
should we comment why we are testing `controlPlane.Machines.Len() < int(*controlPlane.KCP.Spec.Replicas)` and the backside to this choice (to keep track of the recent discussion around this point)?

Reply:
What do you mean the backside of this choice?

Reply:
That remediation is triggered only when we reach the desired number of replicas, see https://docs.google.com/spreadsheets/d/1wakwdk3Y_WDhbb0AfQMC8QxDwPfkSTAmDwMAGy9LUzo/edit#gid=886813936 / 5 Replicas - first provisioning

Reply:
I still don't follow, why would we freeze on 4 replicas on that scenario?

Reply:
Let's assume I'm provisioning a 5 nodes control plane, a problem happens while provisioning the 4th replica, but remediation won't be triggered because of  `controlPlane.Machines.Len() < int(*controlPlane.KCP.Spec.Replicas)` being 4<5.
This is a downside of the current approach, it should be documented, but IMOit is acceptable because it prevents excessive remediation when e.g. I have 5 nodes control plane and two nodes are unhealthy.

Reply:
Ah yes this makes sense. I moved the comments and the logic to be inside `RemediationAllowed`.